plugins/workspace-plugin/src/workspace-folder-updater.ts

```diff
@@ -76,6 +76,15 @@ export class WorkspaceFolderUpdater {
     });
   }
 
+  async removeWorkspaceFolder(path: string): Promise<void> {
```

Comment:
From my point of view the current implementation of this method is not reliable solution.
Please pay attention on [the documentation](https://github.com/eclipse-theia/theia/blob/8b9fb43c763f2f1d179f9839ef993d7218d99073/packages/plugin/src/theia.d.ts#L5913) for the `theia.workspace.updateWorkspaceFolders()`.

Reply:
I tested one of the use cases and can confirm that we should improve the current implementation:

![remove_bug](https://user-images.githubusercontent.com/5676062/114883874-f9582900-9e0d-11eb-8528-dbbca6c8b056.gif)

Reply:
It's not clear for me what's wrong here.

To remove folder `theia.workspace.updateWorkspaceFolders()` receives the folder index (position) for the first argument, and amount of folders to be removed for the next argument, in our case it's one folder.

Could you show an example of more reliable solution?

Reply:
> I tested one of the use cases and can confirm that we should improve the current implementation:


Nice catch here. I will debug it a bit.

Reply:
> The entity should call theia.workspace.updateWorkspaceFolders() then wait for onDidChangeWorkspaceFolders event and only then call theia.workspace.updateWorkspaceFolders() again.


Is this behavior specified somewhere?

From my point of view it's normal if the plugin calls `theia.workspace.updateWorkspaceFolders()` when it's needed.
If the used removes several folders and the plugin receives several events, it's OK to call `theia.workspace.updateWorkspaceFolders()` several times.

Reply:
> Is this behavior specified somewhere?
:-)
Did you see my reference above for the corresponding documentation?

Reply:
> Is this behavior specified somewhere?

Yes it is: https://code.visualstudio.com/api/references/vscode-api#workspace

Reply:
> But I think the reason not in implementation of removeWorkspaceFolder method

please read the documentation, the reference I provided above

Reply:
It it important to be attentive :+1:

> **Note:** it is not valid to call `updateWorkspaceFolders()` multiple times
without waiting for the `onDidChangeWorkspaceFolders()` to fire.

I hadn't written the documentation till the end.